scipy/stats/_continuous_distns.py

```diff
@@ -5508,67 +5508,119 @@ def _entropy(self, s):
         this function uses explicit formulas for the maximum likelihood
         estimation of the log-normal shape and scale parameters, so the
         `optimizer`, `loc` and `scale` keyword arguments are ignored.
+        If the location is free, a likelihood maximum is found by
+        setting its partial derivative wrt to location to 0, and
+        solving by substituting the analytical expressions of shape and
+        scale (or provided parameters). See, e.g., equation 3.1 in [1]_.
+
+        References
+        ----------
+        .. [1] Cohen, A.C., & Whitten, B.J. (1980). Estimation in the
+               three-parameter lognormal distribution, Journal of the
+               American Statistical Association, 75(370), 399-404.
+               https://doi.org/10.2307/2287466
         \n\n""")
     def fit(self, data, *args, **kwds):
-        floc = kwds.get('floc', None)
-        if floc is None:
-            # fall back on the default fit method.
+        alpha_std = kwds.pop('alpha_std', 400)
+        n_steps = kwds.pop('n_steps', 16)
+        xtol = kwds.pop('xtol', None)
```

Comment:
I added these three parameters.

* `alpha_std` controls the width of an interval to investigate;
* `n_steps` one controls the discretisation of this interval;
* `xtol` is passed to `root_scalar` and controls the termination of the root finding.

Tweaking them may impact the result, so I feel like it would be good to not have them set in stone (I will document them in the docstring if you agree with the method).